jkube-kit/generator/api/src/main/java/org/eclipse/jkube/generator/api/DefaultGeneratorManager.java

```diff
@@ -59,6 +60,8 @@ public List<ImageConfiguration> generateAndMerge(List<ImageConfiguration> unreso
     for (ImageConfiguration imageConfiguration : filteredImages) {
       imageConfiguration.setName(imageNameFormatter.format(imageConfiguration.getName()));
       if (imageConfiguration.getBuild() != null) {
+        BuildConfiguration updatedBuildConfig = mergeGlobalConfigParamsWithSingleImageBuildConfig(imageConfiguration.getBuild());
+        imageConfiguration.setBuild(updatedBuildConfig);
```

Comment:
In a future iteration we'll need to see if we can refactor ImageConfiguration.initAndValidate method to take care of this.